client/src/pages/GroupHome.js

```diff
@@ -0,0 +1,218 @@
+import { useUser, useAlert } from '../context/AppContext';
+import {
+  Typography,
+  Grid,
+  Paper,
+  CircularProgress,
+  IconButton,
+  Button,
+  Modal,
+  Box } from '@mui/material';
+import { GridBox } from '../components/GridBox';
+import SettingsIcon from '@mui/icons-material/Settings';
+import EnhancedTable from '../components/BaseTable';
+import BaseGraph from '../components/BaseGraph';
+import { styled } from '@mui/material/styles';
+import {
+  fetchGroupUsers,
+  fetchGroupChartLogs,
+  fetchGroupLogs,
+  fetchGroupSettings,
+  fetchGroupPenalties,
+  postNewGroupMember } from '../lib/api';
+import { groupLogHeaders, groupSettingsHeaders, groupPenaltiesHeaders } from '../lib/tables-config';
+import dayjs from 'dayjs';
+import dayjsPluginUTC from 'dayjs-plugin-utc';
+import weekOfYear from 'dayjs/plugin/weekOfYear';
+
+dayjs.extend(weekOfYear);
+dayjs.extend(dayjsPluginUTC);
+
+async function loadGroupUsers(groupId, setGroupUsers) {
+  const response = await fetchGroupUsers(groupId);
+  setGroupUsers(response);
+}
+
+async function loadGroupChartLogs(userId, setGroupChartLogRows) {
+  const response = await fetchGroupChartLogs(userId);
+  response.forEach((row) => row.date = dayjs(row.date).local().format('MM/DD/YY'));
+  setGroupChartLogRows(response);
+}
+
+async function loadGroupLogs(groupId, setGroupLogRows) {
+  const response = await fetchGroupLogs(groupId);
+  response.forEach((row) => {
+    row.date = dayjs(row.date).local().format('MM/DD/YY')
+  });
+  setGroupLogRows(response);
+}
+
+async function loadGroupSettings(groupId, setGroupSettingsRows) {
+  const response = await fetchGroupSettings(groupId);
+  setGroupSettingsRows(response);
+}
+
+async function loadGroupPenalties(groupId, setGroupPenaltiesRows) {
+  const response = await fetchGroupPenalties(groupId);
+  response.forEach((row) => {
+    row.date = dayjs(row.date).local().format('MM/DD/YY')
+  });
+  setGroupPenaltiesRows(response);
+}
+
+export default function GroupHome() {
+  const { groupId, inviteLink } = useParams();
+  const { user } = useUser();
+  const [groupUsers, setGroupUsers] = useState();
+  const [groupChartLogRows, setGroupChartLogRows] = useState();
+  const [groupLogRows, setGroupLogRows] = useState();
+  const [groupSettingsRows, setGroupSettingsRows] = useState();
+  const [groupPenaltiesRows, setGroupPenaltiesRows] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const { setAlert } = useAlert();
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(!!inviteLink);
+
+  useEffect(() => {
+    if(!user) return navigate('/sign-in');
+    Promise.all([loadGroupUsers(groupId, setGroupUsers),
+        loadGroupLogs(groupId, setGroupLogRows),
+        loadGroupSettings(groupId, setGroupSettingsRows),
+        loadGroupPenalties(groupId, setGroupPenaltiesRows),
+        loadGroupChartLogs(groupId, setGroupChartLogRows)])
+      .then(() => setIsLoading(false))
```

Comment:
You probably want to set loading false in `.finally` instead, since if you get an error you want to display that instead of a loading message.